src/mh-core.jl

```diff
@@ -141,6 +141,22 @@ end
 transition(sampler, model, params) = transition(model, params)
 transition(model, params) = Transition(model, params)
 
+# Called to update proposal when the first sample is drawn
+trackstep!(proposal::Proposal, params) = nothing
+trackstep!(proposal::AbstractArray, params) = foreach(trackstep!, proposal, params)
+trackstep!(proposal::NamedTuple, params) = foreach(trackstep!, proposal, params)
+
+# Called to update proposal when a new step is performed
+# The last argument determines if the step is an acceptance step
+trackstep!(proposal::Proposal, params, 
+           ::Union{Val{true}, Val{false}}) = nothing
+
+trackstep!(proposal::AbstractArray, params, accept::Union{Val{true},Val{false}}) = 
+           foreach((prop, par) -> trackstep!(prop, par, accept), proposal, params)
+
+trackstep!(proposal::NamedTuple, params, accept::Union{Val{true},Val{false}}) = 
+           foreach((prop, par) -> trackstep!(prop, par, accept), proposal, params)
```

Comment:
Can we remove the type annotations on `accept`? They are not needed.
```suggestion
trackstep!(proposal::Proposal, params, accept) = nothing
function trackstep!(proposal::AbstractArray, params, accept)
    return foreach((prop, par) -> trackstep!(prop, par, accept), proposal, params)
end
function trackstep!(proposal::NamedTuple, params, accept)
    return foreach((prop, par) -> trackstep!(prop, par, accept), proposal, params)
end
```

Additionally, again the `NamedTuple` implementation is not completely correct.